src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CngKey.EC.cs

```diff
@@ -26,8 +26,9 @@ internal static bool IsECNamedCurve(string algorithm)
         {
             if (IsECNamedCurve())
             {
-                oidValue = null;
-                return _keyHandle.GetPropertyAsString(KeyPropertyName.ECCCurveName, CngPropertyOptions.None);
+                string? curveName = _keyHandle.GetPropertyAsString(KeyPropertyName.ECCCurveName, CngPropertyOptions.None);
+                oidValue = curveName is null ? null : OidLookup.ToOid(curveName, OidGroup.PublicKeyAlgorithm, fallBackToAllGroups: false);
```

Comment:
Nothing really changed for the return value here. What's changing is the `oidValue` out parameter. Previously it was never getting set for named curves. Now we are just doing an extra "set the `oidValue` out parameter in addition to returning `curveName`".

> And what happens to curve25519 here, since it's a name-with-no-OID on Windows?

It continue to have a null `out oidValue`, but I can add a test for it if you'd like.

Reply:
Pre-coffee I read that as a re-assignment (`curveName = curveName is null ? ...`).  Now I get it.